packages/core/src/types.ts

```diff
@@ -1783,13 +1783,27 @@ export type EmittedFrom<T> = ReturnTypeOrValue<T> extends infer R
   : never;
 
 type ResolveEventType<T> = ReturnTypeOrValue<T> extends infer R
-  ? R extends StateMachine<infer _, infer __, infer TEvent, infer ____>
+  ? R extends StateMachine<
+      infer _,
```

Comment:
I've [experimented a little bit](https://www.typescriptlang.org/play?#code/C4TwDgpgBAsg9gEwgGwDwBUA0UCiA+KAXigG8AoKKAeiqgGM4A7YAJwEMA3NlgSzeYpQAzhEYIAXFAAUEDqOCScASiIEOcHgkE16cLr37BBDZhAAeCqOjIBfMmR2hIUAOIRgAYSbBzwDAWJ0KF9RBCFYRBRUHkYAMwgWKAAlbEZZBIIAfmSoSTS5FgdaJ2g3YBw5Zn8iK2CLUPD4JDQAV0YAa0Y4AHdGbBj4xPwobJxcqHyEshLXdy9TC2rAup8xRsi0AYTk7H4QLJy89MKZsor5JdqQtYjm1D3+uO3h0fHJwrJzMDgWYCgZpIQcLEMrzHyLJpREi6NqWRgtAC2ACNtjZUsc8HgipQcbiAHqZexfH5-AFAgBMNVB3l8qEhaGhDFheURKMSaKge0x2Nx+MJnzM31+-3A0EBQgAzFS5jSIRtUIy4MyJqzUdhoSVJAByACSADkPFqoDZuTpeTiCUTBSSRc5xQAWaXlSp+emod7q20QbX6w3G020c2US0CoWk0XJIEAVid5yqboepC9PoNRpNWLNQZDxOFZKEADZYy66fLFcr4ci1UnNVBdan-RnA1n+TMAPLAAAWCTdWFwAVI2loMX0fAElBEYkkMhdihUhDUGi0lAA+ulmIpB7oR4ZjLLLNY7NjTu5212WGDaeh+0FrmEoKfu-KtokUpzGPsRocJscjxGyg-zz3S4b3qG4ALdZ8digNpOh6RgDiSN4f0cP8T07BIL0WK8ahA1Y73Ap8nhfdECgQpCCmmVDgAAzC-Gw5Zb3CAi7kg19IOXMijgo0MbTzAB2J0aKA5ioRhdcVUrdkSIyRsg2Dfkc3DO0gQADkE9DAIWPwRIZMS4VVKS332WS5Oza1cwjcUAE51LPWjUB0hU9JZSTjU9Gs6z9dMeWbIA) with this and it seems that:
- `any`/`unknown` work for covariant generics
- `never` works work contravariant generic
- only `infer _` works for an invariant generic

Since it's super hard to know the variance type for a given generic... it's best, for now, to just use those placeholders 🤷‍♂️